test/python/algorithms/test_linear_solvers.py

```diff
@@ -0,0 +1,250 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the quantum linear system solver algorithm."""
+
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from scipy.linalg import expm
+import numpy as np
+from ddt import ddt, idata, unpack
+from qiskit import QuantumCircuit
+from qiskit.algorithms.linear_solvers.hhl import HHL
+from qiskit.algorithms.linear_solvers.matrices.tridiagonal_toeplitz import TridiagonalToeplitz
+from qiskit.algorithms.linear_solvers.matrices.numpy_matrix import NumpyMatrix
+from qiskit.algorithms.linear_solvers.observables.absolute_average import AbsoluteAverage
+from qiskit.algorithms.linear_solvers.observables.matrix_functional import MatrixFunctional
+from qiskit.circuit.library.arithmetic.exact_reciprocal import ExactReciprocal
+from qiskit.quantum_info import Operator, partial_trace
+from qiskit.opflow import I, Z, StateFn
+from qiskit import quantum_info
+
+
+@ddt
+class TestMatrices(QiskitAlgorithmsTestCase):
+    """Tests based on the matrices classes.
+
+    This class tests
+        * the constructed circuits
+    """
+
+    @idata([
+        [TridiagonalToeplitz(2, 1, -1 / 3)],
+        [TridiagonalToeplitz(3, 2, 1), 1.1, 3]
+    ])
+    @unpack
+    def test_tridiagonal_toeplitz(self, matrix, time=1.0, power=1):
```

Comment:
It looks like this test does exactly the same as the one below, is that right? If yes, can we just merge them and add one more element in `idata`?

Reply:
yes, merged